data-processor/src/main/java/io/micronaut/data/processor/visitors/finders/criteria/UpdateCriteriaMethodMatch.java

```diff
@@ -170,12 +170,12 @@ protected <T> Predicate interceptPredicate(MethodMatchContext matchContext,
         final SourcePersistentEntity rootEntity = (SourcePersistentEntity) root.getPersistentEntity();
         Predicate predicate = null;
         if (entityParameter != null) {
-            if (rootEntity.getVersion() != null) {
+            if (rootEntity.getVersion() != null && existingPredicate == null) {
```

Comment:
It appears this is used for `void update(Entity entity)` too but also for `void updateByField(String field, Entity entity)` so when there is existing predicate with entity parameter we shouldn't add id and version criteria.